resources/assets/components/utilities/ScholarshipCard/GalleryBlockSignup.js

```diff
@@ -0,0 +1,49 @@
+import gql from 'graphql-tag';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
+import SecondaryButton from '../Button/SecondaryButton';
+
+export const CREATE_SIGNUP_MUTATION = gql`
+  mutation CampaignSignupMutation($campaignId: Int!) {
+    createSignup(campaignId: $campaignId) {
+      id
+    }
+  }
+`;
+const GalleryBlockSignup = ({ campaignId, path }) => {
+  const [handleSignup, { called, error }] = useMutation(
+    CREATE_SIGNUP_MUTATION,
+    {
+      variables: { campaignId },
+    },
+  );
+
+  if (called && !error) {
+    return (window.location = { path });
```

Comment:
I think it would be neat to return a `Spinner` component, now that the user will have to wait for a page refresh. It should make the UX seamless, so it feels like the signup is 'processing' while the campaign page loads! (We also need to assign `path` as a string value):
```suggestion
    window.location = path;
    return <Spinner className="flex justify-center p-4" />
```

We'll need to import the Spinner component as well.

Reply:
makes perfect sense, added! thanks @mendelB I assigned `path` as a string value in props validation. Is that what you meant?

Reply:
Awesome! I was referring to the value we're assigning to the `window.location` which now looks perfect!